updated api, will commit the rest once implement a fix

KS2499 opened the pull request:
Changes:
- Modified models.User, Task and Project
- Split controller to user/project/task controller separately
- Objects hold object ID of other objects instead now
- Added Project Create and Project Get
- Added Task Create, Task Get, Task Modify and Task Delete (Tagging to be implemented)

seetohjinwei commented on api.md:
```diff
@@ -266,71 +266,157 @@ Output:
 
+### Create Project
+
+POST "/project_create"
```
need to return project's id as well (otherwise frontend will never know the id)

-> we might need to add the invite code to the Project model & return it too in project_create as well & implement handlers for joining project

seetohjinwei commented on api.md:
```diff
@@ -266,71 +266,157 @@ Output:
+
+### Add User to Task
+
+PATCH "/task_add_user"
```
-> we need a task_patch/task_modify as well (potentially merge with task_add_user or maybe just keep them separate)
-> this is used for both task editing and task drag & dropping (probably have to add dnd_id into Task model on server side as well)

seetohjinwei commented on api.md:
```diff
@@ -266,71 +266,157 @@ Output:
+
+Status Code: 200 or 400
+
+### Create Task
```
-> i think we need a task_get_all & event_get_all (or just data_get_all that does both), even though events not implemented yet just return empty or something
-> so that we can seed the data

seetohjinwei commented on api.md:
```diff
@@ -266,71 +266,157 @@ Output:
+
+### Get Project
+
+GET "/project_get"
```
need to return
1. members
2. events
3. tasks

seetohjinwei commented on api.md:
```diff
@@ -266,71 +266,157 @@ Output:
+
+### Create Task
+
+POST "/task_create"
```
-> task_create has to return the task's id as well
-> needs to take in deadline as well
-> maybe rename users to assignedTo for consistency

seetohjinwei commented on api.md:
````diff
@@ -266,71 +266,157 @@ Output:
+  tasks: string[];
+}
+```
+
````
misc stuff:

-> need to write the exact outputs (otherwise idk the exact name) of the requests (similarly to the inputs)

-> also do you want to handle creationTime on server side or client side? (i assume its server side because you don't take it in?)

KS2499 replied:
Yah creation time is handled server side

KS2499 commented:
Changes:
- Modified models.User, Task and Project
- Split controller to user/project/task controller separately
- Objects hold object ID of other objects instead now
- Added Project Create and Project Get
- Added Task Create, Task Get, Task Modify and Task Delete (Tagging to be implemented)
